### PR TITLE
{lib,tools}[foss/2023a] Lightning v2.2.1 w/ CUDA 12.1.1

### PR DESCRIPTION
**Pull request by NPotvin**
(created using `eb --new-pr`)
requires:
- #19648

### CODE REVIEW

**Comment by NPotvin** [comment]
To make it work, I used `--from-pr 19666,19648` for the ECs of PyTorch, deepdiff and tensordboardX

File: easybuild/easyconfigs/l/lightning/lightning-2.2.0.post0-foss-2023a-PyTorch-2.1.2-CUDA-12.1.1.eb [hunk 1]
```diff
@@ -0,0 +1,28 @@
+easyblock = 'PythonPackage'
+
+name = 'lightning'
```
Comment (boegel) [inline easybuild/easyconfigs/l/lightning/lightning-2.2.0.post0-foss-2023a-PyTorch-2.1.2-CUDA-12.1.1.eb:3]: @NPotvin Isn't this basically the same thing as `PyTorch-Lightning-2.2.0.post0-foss-2023a-CUDA-12.1.1.eb`?

Or is there actually something additional installed here on top of `PyTorch-Lightning`?

File: easybuild/easyconfigs/d/deepdiff/deepdiff-6.7.1-GCCcore-12.3.0.eb [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+version = '6.7.1'
+
+homepage = 'https://zepworks.com/deepdiff/current/'
+description = """DeepDiff: Deep Difference of dictionaries, iterables and almost any other object recursively."""
+
+toolchain = {'name': 'GCCcore', 'version': '12.3.0'}
+
+builddependencies = [
+    ('binutils', '2.40'),
+    ('hatchling', '1.18.0'),  # for setuptools-scm
```
Comment (boegel) [inline easybuild/easyconfigs/d/deepdiff/deepdiff-6.7.1-GCCcore-12.3.0.eb:13]: No longer needed thanks to #19777

**Review by boegel — DISMISSED** [review]

File: easybuild/easyconfigs/l/lightning/lightning-2.2.0.post0-foss-2023a-PyTorch-2.1.2-CUDA-12.1.1.eb (hunk 1 above)
Comment (NPotvin) [reply easybuild/easyconfigs/l/lightning/lightning-2.2.0.post0-foss-2023a-PyTorch-2.1.2-CUDA-12.1.1.eb:3]: @boegel It's an additional layer on top of PyTorch lightning. It is developed by the same team, apparently. On pypi lightning and pytorch-lightning are two different packages.
(have a look [here](https://github.com/Lightning-AI/pytorch-lightning))

File: easybuild/easyconfigs/d/deepdiff/deepdiff-6.7.1-GCCcore-12.3.0.eb (hunk 2 above)
Comment (NPotvin) [reply easybuild/easyconfigs/d/deepdiff/deepdiff-6.7.1-GCCcore-12.3.0.eb:13]: I see another deepdiff here #19648 (to be fair, I think I copied files from there to adapt them to CUDA and forgot deepdiff and tensorboardX were not from me)
But I don't see how #19777 would solve the requirement for deepdiff. If I remove deepdiff and tensorboardX from this PR and leave Pytorch-Lightning as is, would that be ok ?

File: easybuild/easyconfigs/l/lightning/lightning-2.2.0.post0-foss-2023a-PyTorch-2.1.2-CUDA-12.1.1.eb (hunk 1 above)
Comment (boegel) [reply easybuild/easyconfigs/l/lightning/lightning-2.2.0.post0-foss-2023a-PyTorch-2.1.2-CUDA-12.1.1.eb:3]: Zero room for confusion there... 🙈

File: easybuild/easyconfigs/d/deepdiff/deepdiff-6.7.1-GCCcore-12.3.0.eb (hunk 2 above)
Comment (smoors) [reply easybuild/easyconfigs/d/deepdiff/deepdiff-6.7.1-GCCcore-12.3.0.eb]: @NPotvin the comment of @boegel refers to the line above it: the `hatchling` dep is no longer needed for `deepdiff`.
but it should actually be removed from his own PR https://github.com/easybuilders/easybuild-easyconfigs/pull/19648 (unless you copy it into this PR).

File: easybuild/easyconfigs/l/lightning/lightning-2.2.0.post0-foss-2023a-PyTorch-2.1.2-CUDA-12.1.1.eb (hunk 1 above)
Comment (smoors) [inline easybuild/easyconfigs/l/lightning/lightning-2.2.0.post0-foss-2023a-PyTorch-2.1.2-CUDA-12.1.1.eb:3]: @NPotvin please use uppercase `Lightning` here, as that seems to be the official name?

**Comment by smoors** [comment]
@NPotvin PyTorch-Lightning has been updated to v2.2.1 in #19648 
i much prefer you use that version as dep for `Lightning` (and probably update it to v2.2.1 also)
or do you have a specific reason for using v2.2.0.post0 ?

**Comment by NPotvin** [comment]
@smoors Thanks for reviewing this PR !
Version 2.2.1 was released last week. That's the reason for 2.2.0.post0.
I won't have the time to deal with it today, but I'll go through your comments and change my PR accordingly, including upgrading to 2.2.1.

**Comment by boegel** [comment]
@smoors Is this good to merge?

**Comment by smoors** [comment]
@NPotvin i removed the PyTorch versionsuffix, as v2.1.2 is already the default version for the 2023a toolchain generation.

**Comment by smoors** [comment]
@boegelbot: please test @ generoso

**Comment by smoors** [comment]
@boegelbot please test @ jsc-zen3

**Review by smoors — APPROVED** [review]
lgtm

**Review by smoors — APPROVED** [review]
lgtm

**Comment by smoors** [comment]
Going in, thanks @NPotvin!

**Pull request merged**